client/keys/add.go

```diff
@@ -117,56 +118,65 @@ func RunAddCmd(ctx client.Context, cmd *cobra.Command, args []string, inBuf *buf
 		return err
 	}
 
-	if dryRun, _ := cmd.Flags().GetBool(flags.FlagDryRun); !dryRun {
-		_, err = kb.Key(name)
-		if err == nil {
-			// account exists, ask for user confirmation
-			response, err2 := input.GetConfirmation(fmt.Sprintf("override the existing name %s", name), inBuf, cmd.ErrOrStderr())
-			if err2 != nil {
-				return err2
-			}
+	_, err = kb.Key(name)
+	if err == nil {
+		// account exists, ask for user confirmation
+		response, err2 := input.GetConfirmation(fmt.Sprintf("override the existing name %s", name), inBuf, cmd.ErrOrStderr())
+		if err2 != nil {
+			return err2
+		}
 
-			if !response {
-				return errors.New("aborted")
-			}
+		if !response {
+			return errors.New("aborted")
+		}
 
-			err2 = kb.Delete(name)
-			if err2 != nil {
-				return err2
-			}
+		if dryRun {
+			return errors.New("override not available with --dry-run")
+		}
+
+		err2 = kb.Delete(name)
+		if err2 != nil {
+			return err2
+		}
+	}
+
+	multisigKeys, _ := cmd.Flags().GetStringSlice(flagMultisig)
+	if len(multisigKeys) != 0 {
+		var pks []cryptotypes.PubKey
```

Comment:
For better efficiency, I would define `pks` as slice with finite cap like that `pks := make([]cryptotypes.PubKey, multisigKeys)`

Reply:
Thanks! I did not write the code for this, I'm just removing the condition that this section was wrapped within. I would like to keep this pull request focused on fixing the issue with using dry run. Making this change also causes a panic on line 162. Maybe we can address optimizations in a separate issue?

Reply:
Thanks! This also causes a panic on line 162 with sorting. Maybe we can optimize in a separate pull request?